### DISCUSSION TITLE
ClassNotFoundException while trying to use an authentication provider provided by a protocol handler

### DISCUSSION THREAD

**Discussion by emmanuelGuiton** [apache/pulsar#20306] [Q&A]
Hi,

I am trying to set up an authentication provider that is provided by a third party protocol handler (StreamNative AMQP on Pulsar). I can successfully set up the protocol handler but I can't figure out how to use the AMQP authentication provider. It always ends in a `ClassNotFoundException: io.streamnative.pulsar.handlers.amqp.authentication.AuthenticationProviderBasic` though the class is present in the unpacked NAR archive. So I am wondering if there is a bug in Pulsar because the classloader ignores a few classes from the protocol handler or if I am simply doing something wrong ... but I can't figure what.

Any help ?

I already reported the issue as a bug here https://github.com/streamnative/aop/issues/864

**Comment by tisonkun** [comment]
Let's leave the third party plugin questions in their own repo.

**Comment by emmanuelGuiton** [comment]
Well : the point of my question was to understand if the issue came from Pulsar ...
